File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
         const checker = {
            "}":"{",
            "]":"[",
            ")":"("
        }
        let stack = [];

        for(let i=0;i<s.length;i++){
            if(checker[s[i]]){
                if(stack.length > 0 && stack[stack.length-1] ===checker[s[i]] ){
                stack.pop()
                }   else{
                return false
                }
                
            }else{
                stack.push(s[i])
            }
        }
        return stack.length === 0
    }
}

File: Data Structures & Algorithms/car-fleet/submission-0.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int [][] pair = new int[position.length][2] ;
        for(int i=0;i<position.length;i++){
            pair[i][0] = position[i];
            pair[i][1] =speed[i] ;
        }

        Arrays.sort(pair,(a,b) ->{
            return Integer.compare(b[0],a[0]);
        });
        System.out.println(Arrays.deepToString(pair));

        Stack<Double> stack =  new Stack<>();

        for(int[] el : pair){
            double time = (double)(target-el[0])/el[1];
            stack.push(time);
            if(stack.size() >=2 && stack.peek() <= stack.get(stack.size()-2)){
                stack.pop();
            }

        }
        return stack.size();
    }
}
